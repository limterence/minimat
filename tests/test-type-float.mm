/* test-type-float: float literals, operators and relationals */
float addme(float x) {   /* float function definition and argument */
  return x + 39.5;
}
int main() {
  float a;
  float b;
  float c;
  a = -2.3;              /* literals */
  b = 4.6 + 3.0;
  printfloat(-a);        /* arithmetic operators */
  printfloat(a);
  printfloat(a + b);
  printfloat(a - 10.9); 
  printfloat(5.0 * b);
  printfloat(a / b);
  printfloat(a % 2.0);
  printfloat(a ^ 3.0);
  printb(a == -2.3);      /* relationals */
  printb(a == b);
  printb(a != b);
  printb(a >= b);
  printb(a > b);
  printb(a <= b);
  printb(a < b);
  printfloat(addme(2.5)); /* function call */
  println();
}
