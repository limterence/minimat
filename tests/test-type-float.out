  2.30  -2.30   5.30 -13.20  38.00  -0.30  -0.30 -12.17 1 
0 
1 
0 
0 
1 
1 
 42.00 
